lib/Notes/NoteForm/NoteForm.js

```diff
@@ -0,0 +1,288 @@
+    onCancel: PropTypes.func.isRequired,
+    onSubmit: PropTypes.func.isRequired,
+    paneHeaderAppIcon: PropTypes.string.isRequired,
+    referredRecord: referredRecordShape.isRequired,
+    submitSucceeded: PropTypes.bool.isRequired,
+  };
+
+  state = {
+    sections: {
+      noteForm: true,
+      assigned: true,
+    }
+  };
+
+  detailsEditorRef = React.createRef();
+
+  handleSectionToggle = ({ id }) => {
+    this.setState(prevState => ({
+      sections: {
+        ...prevState.sections,
+        [id]: !prevState.sections[id],
+      }
+    }));
+  }
+
+  handleToggleAll = (toggledSections) => {
+    this.setState({
+      sections: toggledSections,
+    });
+  }
+
+  renderExpandAllButton = () => {
+    return (
+      <ExpandAllButton
+        accordionStatus={this.state.sections}
+        onToggle={this.handleToggleAll}
+      />
+    );
+  }
+
+  renderNoteMetadata = () => {
+    const { noteMetadata } = this.props;
+
+    return (
+      <Row>
+        <Col xs={12}>
+          <MetaSection {...noteMetadata} />
+        </Col>
+      </Row>
+    );
+  }
+
+  renderFirstMenu = () => {
+    const { onCancel } = this.props;
+
+    return (
+      <IconButton
+        icon="times"
+        onClick={onCancel}
+      />
+    );
+  }
+
+  renderLastMenu = (formIsPristine) => {
+    return (
+      <Button
+        type="submit"
+        buttonStyle="primary"
+        marginBottom0
+        disabled={formIsPristine}
+      >
+        <FormattedMessage id="stripes-smart-components.notes.saveAndClose" />
+      </Button>
+    );
+  }
+
+  getActionMenu = ({ onToggle }) => {
+    const { onCancel } = this.props;
+
+    return (
+      <Fragment>
+        <Button
+          buttonStyle="dropdownItem"
+          onClick={() => {
+            onCancel();
+            onToggle();
+          }}
+        >
+          <Icon icon="times-circle">
+            <FormattedMessage id="stripes-smart-components.notes.cancel" />
+          </Icon>
+        </Button>
+      </Fragment>
+    );
+  };
+
+  validateFormFields = (values) => {
+    const { title } = values;
+    const errors = {};
+
+    if (this.detailsEditorRef.current) {
+      const detailsPlainText = this.detailsEditorRef.current.getEditor().getText();
+
+      errors.content = this.getDetailsValidationError(detailsPlainText);
+    }
+
+    errors.title = this.getTitleValidationError(title);
+
+    return errors;
+  }
+
+  getDetailsValidationError = (details) => {
+    if (details.length > MAX_DETAILS_LENGTH) {
+      return this.renderCharacterLimitExceededError('details');
+    }
+
+    return null;
+  }
+
+  getTitleValidationError(title) {
+    if (!title) {
+      return <FormattedMessage id="stripes-smart-components.notes.title.missing" />;
+    }
+
+    if (title.length > MAX_TITLE_LENGTH) {
+      return this.renderCharacterLimitExceededError('title');
+    }
+
+    return null;
+  }
+
+  renderCharacterLimitExceededError = (fieldName) => {
```

Comment:
I do not think arrow function is needed here since `this` is not used. Check other places.